[RFC 54] Internationalisation reference docs (#6377)
wagtail/wagtail: docs/reference/settings.rst

Set the number of days (default 7) that search query logs are kept for; these are used to identify popular search terms for :ref:`promoted search results <editors-picks>`. Queries older than this will be removed by the :ref:`search_garbage_collect` command.

Internationalisation
====================

.. versionadded:: 2.11

Wagtail supports internationalisation of content by maintaining separate trees
of pages for each language.

For a guide on how to enable internationalisation on your site, see the :ref:`configuration guide <enabling_internationalisation>`.

``WAGTAIL_I18N_ENABLED``
------------------------

(boolean, default ``False``)

When set to ``True``, Wagtail's internationalisation features will be enabled.

For example:

.. code-block:: python

    WAGTAIL_I18N_ENABLED = True

``WAGTAIL_CONTENT_LANGUAGES``
-----------------------------

> (list, default [])

Reviewer: ```suggestion
(list, default ``[]``)
```
to be in line with the above ``False``

A list of languages that Wagtail content can be authored in.

This setting follows the same format that Django uses for its ``LANGUAGES`` setting,
so you can set them both to the same value:

.. code-block:: python

    LANGUAGES = WAGTAIL_CONTENT_LANGUAGES = [
        ('en-GB', _("English (United Kingdom)")),
        ('en-US', _("English (United States)")),
        ('es-ES', _("Spanish (Spain)")),
        ('es-MX', _("Spanish (Mexico)")),
    ]

> But having them separate allows you to configure many different regions on your site
> but have them share Wagtail content (but defer on things like date formatting, currency, etc):

Reviewer: ```suggestion
However having them separate allows you to configure many different regions on your site
yet have them share Wagtail content (but defer on things like date formatting, currency, etc):
```

```suggestion
yet have them share Wagtail content (but defer on things like date formatting, currency, etc):
```
"But having them...but have them..."  sounds weird

.. code-block:: python

    LANGUAGES = [
        ('en', _("English (United Kingdom)")),
        ('en-US', _("English (United States)")),
        ('es', _("Spanish (Spain)")),
        ('es-MX', _("Spanish (Mexico)")),
    ]

    WAGTAIL_CONTENT_LANGUAGES = [
        ('en', _("English")),
        ('es', _("Spanish")),
    ]

> This would mean that your site will respond on the URLs
> ``https://www.mysite.com/es/`` and ``https://www.mysite.com/es-MX/`` but both
> of these URLs will serve content from the same "Spanish" tree in Wagtail.

Reviewer: ```suggestion
This would mean that your site will respond on the
``https://www.mysite.com/es/`` and ``https://www.mysite.com/es-MX/`` URLs, but both
of them will serve content from the same "Spanish" tree in Wagtail.
```

Set the number of days (default 7) that search query logs are kept for; these are used to identify popular search terms for :ref:`promoted search results <editors-picks>`. Queries older than this will be removed by the :ref:`search_garbage_collect` command.

Internationalisation
====================

.. versionadded:: 2.11

Wagtail supports internationalisation of content by maintaining separate trees
of pages for each language. This feature is decribed more fully in TODO

The internationalisation feature is configured with two settings:

``WAGTAIL_I18N_ENABLED``

(boolean, default False)

When set to True, Wagtail's internationalisation features will be enabled.

For example:

.. code-block:: python

    WAGTAIL_I18N_ENABLED = True

``WAGTAIL_CONTENT_LANGUAGES``

(list, default [])

A list of languages that Wagtail content can exist in. This list is in the

> exact same format as Django’s ``LANGUAGES`` setting. It’s separate from that

Reviewer 2: exact same format -> same format

> exact same format as Django’s ``LANGUAGES`` setting. It’s separate from that

Reviewer 2: It’s -> It's

        ('en', _("English (United Kingdom)")),
        ('en-US', _("English (United States)")),
        ('es', _("Spanish (Spain)")),
        ('es-MX', _("Spanish (Mexico)")),
    ]

    WAGTAIL_CONTENT_LANGUAGES = [
        ('en', _("English")),
        ('es', _("Spanish")),
    ]

This would mean that your site will respond on the URLs
``https://www.mysite.com/es/`` and ``https://www.mysite.com/es-MX/`` but both
of these URLs will serve content from the same Spanish tree in Wagtail.

You might want to do this if you plan to implement some form of programmatic
localisation (such as changing the default currency on a donation form, for
example) without having to store two copies of the content in Wagtail.

The format of this setting is intentionally the same as ``LANGUAGES`` so you
can set them both at the same time if they have the same value:

.. code-block:: python

    LANGUAGES = WAGTAIL_CONTENT_LANGUAGES = [
        ('en-GB', _("English (United Kingdom)")),
        ('en-US', _("English (United States)")),
        ('es-ES', _("Spanish (Spain)")),
        ('es-MX', _("Spanish (Mexico)")),
    ]

.. note:: ``WAGTAIL_CONTENT_LANGUAGES`` must be a subset of ``LANGUAGES``

>     It's important to note that all languages that exist in ``WAGTAIL_CONTENT_LANGUAGES``

Reviewer 2: It's important to note that -> Note that

Set the number of days (default 7) that search query logs are kept for; these are used to identify popular search terms for :ref:`promoted search results <editors-picks>`. Queries older than this will be removed by the :ref:`search_garbage_collect` command.

Internationalisation
====================

.. versionadded:: 2.11

Wagtail supports internationalisation of content by maintaining separate trees
of pages for each language.

For a guide on how to enable internationalisation on your site, see the :ref:`configuration guide <enabling_internationalisation>`.

``WAGTAIL_I18N_ENABLED``
------------------------

(boolean, default ``False``)

When set to ``True``, Wagtail's internationalisation features will be enabled.

> For example:

Author: Removed, thanks!

    LANGUAGES = WAGTAIL_CONTENT_LANGUAGES = [
        ('en-gb', _("English (United Kingdom)")),
        ('en-us', _("English (United States)")),
        ('es-es', _("Spanish (Spain)")),
        ('es-mx', _("Spanish (Mexico)")),
    ]

However having them separate allows you to configure many different regions on your site
yet have them share Wagtail content (but defer on things like date formatting, currency, etc):

.. code-block:: python

    LANGUAGES = [
        ('en', _("English (United Kingdom)")),
        ('en-us', _("English (United States)")),
        ('es', _("Spanish (Spain)")),
        ('es-mx', _("Spanish (Mexico)")),
    ]

    WAGTAIL_CONTENT_LANGUAGES = [
        ('en', _("English")),
        ('es', _("Spanish")),
    ]

This would mean that your site will respond on the
``https://www.mysite.com/es/`` and ``https://www.mysite.com/es-MX/`` URLs, but both
of them will serve content from the same "Spanish" tree in Wagtail.

.. note:: ``WAGTAIL_CONTENT_LANGUAGES`` must be a subset of ``LANGUAGES``

    Note that all languages that exist in ``WAGTAIL_CONTENT_LANGUAGES``
    must also exist in your ``LANGUAGES`` setting. This is so that Wagtail can

>     generate a live URL to these pages from an untranslated context (eg, the admin

Reviewer 3: ```suggestion
    generate a live URL to these pages from an untranslated context (e.g. the admin
```
